### ISSUE TITLE
X server crashed with example running mutiprocessing

### ISSUE THREAD

**Issue by dealmeidavf** [dpploy/cortix#185]
https://github.com/dpploy/cortix/blob/dc1b401bba1a6e9211b30703a413006a517c3cb5/cortix/examples/fluidflow/run_droplet_multi_plot.py#L33

When running this example with `use_mpi=False`, I get fatal IO errors on my X server. In fact at some point it crashed my machine and corrupted the filesystem.
Could someone verify this example is problematic when running with multiprocessing and why?
Thanks.

Output: 
[snip]
Vortex::time[s] = 0.0
Vortex::time[s] = 10.0
Vortex::time[s] = 20.0
XIO:  fatal IO error 25 (Inappropriate ioctl for device) on X server ":0"
      after 426 requests (426 known processed) with 15 events remaining.
XIO:  fatal IO error 25 (Inappropriate ioctl for device) on X server ":0"
      after 426 requests (426 known processed) with 15 events remaining.
XIO:  fatal IO error 25 (Inappropriate ioctl for device) on X server ":0"
      after 434 requests (434 known processed) with 19 events remaining.
Destroyed Cortix object

**Comment by tazzaoui** [comment]
I am guessing you are hitting the limit on the number of processes you can fork on your machine. Maybe we can look into pooling for keeping the number of spawned processes down

**Comment by dealmeidavf** [comment]
This may be a matplotlib issue (what else uses the X server?)? Why would I have more than 8 processes open on multiprocess mode?

**Comment by dealmeidavf** [comment]
I think the problem is as follows. For the multiple files plotting example case with 5 droplets, there are 25 instances of the DataPlot module created. Because of the duplication of module instantiation on all processes and the corresponding import of matplotlib, I am guessing each import of matplotlib with backend etc. makes an X server IO request. There are only so many of those any system can serve. Obviously, plotting on the fly is something we do not want to support in parallel computing. However it pointed out the issues of Cortix assembling duplicates of module instances.

**Comment by tazzaoui** [comment]
The example in your branch runs fine on my machine. 

In theory multiprocessing should be fine for matplotlib, it is multi-threading where matplotlib issues occur. Either way, it might be more efficient to have one DataPlot module serving all of the droplets. Creating one for each seems to be unnecessary for most simulations.

**Comment by dealmeidavf** [comment]
There is some issue with multiprocessing since this does not happen with MPI on my machine.

**Comment by tazzaoui** [comment]
Please post the contents of your Xorg log file  (`~/.local/share/xorg/Xorg.0.log`) so we can debug this issue

**Comment by dealmeidavf** [comment]
I commented this line

https://github.com/dpploy/cortix/blob/ddb5a1002b61afce8bf08fe23118dc88839f793d/cortix/examples/run_droplet.py#L80

and no problem with the X server on my machine. Something in that function is making a lot of requests for X resources. Obviously not all processes need to plot the same thing...For now this stops crashing my machine but it reveals an important point of plotting in parallel.

**Comment by dealmeidavf** [comment]
Unfortunately the crash is still there whether or not the `vortex.plot_velocity()` is used or not...problem is still there...

**Comment by dealmeidavf** [comment]
> Please post the contents of your Xorg log file (`~/.local/share/xorg/Xorg.0.log`) so we can debug this issue

I do not have /xorg in [snip]/share. Maybe /var/log? I will check.
